core/src/main/java/org/elasticsearch/search/suggest/Suggest.java

```diff
@@ -522,18 +565,55 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
                 return builder;
             }
 
+            public static Entry<Option> fromXContent(XContentParser parser) throws IOException {
+                Token token = parser.currentToken();
+                ensureExpectedToken(XContentParser.Token.START_OBJECT, token, parser::getTokenLocation);
+                ensureFieldName(parser, parser.nextToken(), Fields.TEXT);
+                ensureExpectedToken(XContentParser.Token.VALUE_STRING, parser.nextToken(), parser::getTokenLocation);
```

Comment:
I decided to rely on the order of how we print fields here for now, can also change that to ConstuctongObjectParser, shouldn't be too difficult here.